native/KotlinIOS/KotlinIOS/Main/MainPresenter.swift

```diff
@@ -13,9 +13,14 @@ struct MainPresenter: MainPresenterProtocol {
         api
             .getUser(userName: "gioevi90")
             .flatMap(f: { self.api.getFollowers(url: $0.followersUrl) as! Request<AnyObject> })
-            .executeCallback(onSuccess: { $0.fold(fe: { error in self.onFetchFollowersError(error: error.errorBody) },
-                                                  fs: { success in self.onFetchFollowersSuccess(followers: success.body) }) },
-                             onError: { self.onFetchFollowersError(error: $0.message ?? "")  })
+            .execute{ result, error in
+               if let e = error {
+                self.onFetchFollowersError(error: e.localizedDescription )
+               } else {
+                result!.fold(fe: { error in self.onFetchFollowersError(error: error.errorBody) },
```

Comment:
⚠️ Avoid force unwrapping

Reply:
I thought it was needed because I saw it in the [example](https://kotlinlang.org/docs/reference/whatsnew14.html#support-for-kotlins-suspending-functions-in-swift-and-objective-c).

Reply:
of course is not... it is better unwrap it before use it :) you can simply add `if let res = result ` next to the else keywork and use res instead of result!

Reply:
> I thought it was needed because I saw it in the [example](https://kotlinlang.org/docs/reference/whatsnew14.html#support-for-kotlins-suspending-functions-in-swift-and-objective-c).

Indeed it is to be avoided.

`result.map { result in result.fold(fe:fs:) }`

Reply:
@dcampogiani ^

Reply:
Sup mate?

Reply:
The answer you requested

Reply:
@Gioevi90 @Faber92 is it fine now?